sentry/src/main/java/io/sentry/ServerNameResolvingEventProcessor.java

```diff
@@ -0,0 +1,134 @@
+package io.sentry;
+
+import io.sentry.util.Objects;
+import java.net.InetAddress;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * An {@link EventProcessor} that sets the local ip address to an {@link
+ * SentryEvent#getServerName()} property.
+ *
+ * <p>Note - this event processor is not used by SDK by default and must be configured manually on
+ * {@link SentryOptions} if there is intention to use it.
+ */
+public final class ServerNameResolvingEventProcessor implements EventProcessor {
```

Comment:
We could add this to the list automatically if `sendDefaultPii` is set to true in Java. In the core SDK I mean. 

Though we should probably remote this on Android. /cc @marandaneto

Reply:
Similar to https://github.com/getsentry/sentry-java/pull/1144#discussion_r550449131 - isn't it better to keep it as an opt-in rather than something our other integrations have to opt-out from?

Reply:
Since we can just use the IP address, maybe opt-in for this one sounds better (since it's not so cheap to get it)

Reply:
on Android, it won't make sense, adding by default would require that the Android integration removes it,  we'd be allocating something that would not be used by the Android integration and this kinda slows down the App. init (if every option would be added by default and removed by Android) which has limited resources.

I'd prefer to add by default in the needed integrations (Spring, etc...), the downside is, when somebody depends on `sentry` only, this integration won't be there.

I guess this trade-off will become more common with the number of features that have been adding to the `sentry` package, wondering if a `flag` on options would make sense, which differs mobile from different environments and based on that, we allocate those objects or not.

Reply:
If opting out would be removing integrations by name or class name I think it could be an option. But currently there's no simple way of opting out from event processor activated by default. 

We could have something like: `SentryOptions#removeEventProcessor(ServerNameResolvingEventProcessor.class)`

But as @marandaneto said, it's too late, objects have already been initialized. I remain in favour of activating things where we are sure they make sense and keeping `sentry` core as the lowest common denominator.

Reply:
Folks at Sentry have strong opinions about default EventProcessors. The idea is that the SDK shouldn't ship any eventprocessor but integrations instead (that dig through the static API to find the client, then the options, to then look inside the integrations list of that options object, and only then do its work or not.

That said, I also prefer not to have such cryptic API like `SentryOptions#removeEventProcessor(ServerNameResolvingEventProcessor.class)`. I rather have proper Options for things when possible.

We can have an option such as `AttachServerName` and if this flag is on, the SDK adds the integration. On Android we can just `new SentryOptions` and set this flat to false so it won't mean any runtime cost.

Reply:
Understood. It's enabled by default now and new flag has been added to `SentryOptions`.